src/pyinstaller_versionfile/__main__.py

```diff
@@ -1,48 +1,167 @@
+        if value is None:
+            keys_to_remove.append(key)
+    for key in keys_to_remove:
+        optional_args.pop(key)
+
+
+    if args.source_format == "yaml":
+        pyinstaller_versionfile.create_versionfile_from_input_file(
+            output_file=args.outfile,
+            input_file=args.metadata_source,
+            **optional_args,
+        )
+    elif args.source_format in ["distribution", "dist"]:
+        pyinstaller_versionfile.create_versionfile_from_distribution(
+            output_file=args.outfile,
+            distname=args.metadata_source,
+            **optional_args,
+        )
+    else:
+        pyinstaller_versionfile.create_versionfile(
+            output_file=args.outfile,
+            **optional_args,
+        )
+
+
+def parse_args_make_version(args):
+    parser = argparse.ArgumentParser(
+        description="Create a version file for PyInstaller."
+    )
+    parser.add_argument(
+        "--metadata-source",
+        help="Required if --source-format is specified. Either path to the input file, or name of the distribution.",
+    )
+    parser.add_argument(
+        "--source-format",
+        choices=["yaml", "distribution", "dist"],
+        help="Define the source format expected in --metadata-source.",
+    )
+    parser.add_argument(
+        "--outfile",
+        default="./version_file.txt",
+        help="Resulting version file for PyInstaller.",
+    )
+    parser.add_argument(
+        "--version",
+        default=None,
+        help="Override Version information given in metadata file.",
+    )
+    parser.add_argument(
+        "--company-name",
+        default=None,
+        help="Name of the company that produced the file.",
+    )
+    parser.add_argument(
+        "--file-description",
+        default=None,
+        help=("Description to be presented to users. "
+              "It may be displayed when the user is choosing files to install."),
+    )
+    parser.add_argument(
+        "--internal-name",
+        default=None,
+        help=("Internal name of the file. "
+              "If the file has no internal name, this string should be the original filename, without extension."),
+    )
+    parser.add_argument(
+        "--legal-copyright",
+        default=None,
+        help=("Copyright notices that apply to the file. "
+              "This should include the full text of all notices, legal symbols, copyright dates, and so on."),
+    )
+    parser.add_argument(
+        "--original-filename",
+        default=None,
+        help=("Original name of the file, not including a path. "
+              "This information enables an application to determine whether a file has been renamed by a user."),
+    )
+    parser.add_argument(
+        "--product-name",
+        default=None,
+        help="Name of the product with which the file is distributed.",
+    )
+
+    # TODO: idea for translation? Maybe langID=0;charsetID=1200? or just <langID>:<charsetID>?
```

Comment:
keep in mind, it is possible to define multiple code pairs. why not just translation, which expect a tuple or a list of tuples?

Reply:
Probably using the `nargs="*"` option of argparse would be best for this, and then convert it accordingly. Then it's possible to pass this option multiple times. 
But I will leave this out of scope for this MR.